### PR TITLE
LPS-162139,LPS-162137

### PR DESCRIPTION
**Pull request by p-albuquerque**

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
@p-albuquerque some of the failures seem to be related. Could you take a look, please?

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Just started reviewing :)

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:test:relevant

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectViewDTOConverter.java [hunk 1]
```diff
@@ -152,36 +157,63 @@ private ObjectViewFilterColumn _toObjectViewFilterColumn(
...
-							itemValue -> _language.get(
-								locale,
-								WorkflowConstants.getStatusLabel(itemValue))),
+							(List<Map<String, Object>>)preloadedData.get(
+								"itemsValues"),
+							itemValue -> {
+								try {
+									return _objectEntryLocalServiceImpl.
+										getTitleValue(
+											(Long)itemValue.get("value"),
```
Comment (marcelabc) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/converter/ObjectViewDTOConverter.java]: check cast (long)

File: modules/apps/object/object-api/src/main/java/com/liferay/object/field/filter/parser/ObjectFieldFilterParser.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 public interface ObjectFieldFilterParser {
 
 	public Map<String, Object> parse(
-			long listTypeDefinitionId, Locale locale,
+			long listTypeDefinitionId, Locale locale, long objectDefinitionId,
```
Comment (marcelabc) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/field/filter/parser/ObjectFieldFilterParser.java:29]: check validate method parameter (long listTypeDefinitionId -> long objectDefinitionId)

Comment (marcelabc) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/field/filter/parser/ObjectFieldFilterParser.java:29]: Sort the parameters in parse()

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 3]
```diff
@@ -115,18 +142,82 @@ public void validate(
...
+
+				_validate(
+					jsonArray, objectDefinitionId,
+					objectViewFilterColumn.getObjectFieldName());
+			}
+		}
+	}
+
+	private List<Map<String, Object>> _toIdsList(
+			JSONArray jsonArray, long objectDefinitionId,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:179]: sort the parameters

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 4]
```diff
@@ -139,7 +230,72 @@ private List<Integer> _toIntegerList(JSONArray jsonArray) {
 		return statuses;
 	}
 
+	private void _validate(
+			JSONArray jsonArray, long objectDefinitionId,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:241]: double check the sort

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 5]
```diff
@@ -139,7 +230,72 @@ private List<Integer> _toIntegerList(JSONArray jsonArray) {
...
+				objectDefinitionId1);
+
+		if (objectDefinition1.isSystem()) {
+			for (int i = 0; i < jsonArray.length(); i++) {
+				PersistedModelLocalService persistedModelLocalService =
+					_persistedModelLocalServiceRegistry.
+						getPersistedModelLocalService(
+							objectDefinition1.getClassName());
+
+				persistedModelLocalService.getPersistedModel(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:266]: Add try/catch

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hey @p-albuquerque,
We reviewed together, but it would be better to ask for a review from @MarinhoFeliphe as well :smile:

**Comment by marcelabc** [comment]
Please, remember to run all integration tests ;)

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:stop

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
Resent here: #1153

**Comment by p-albuquerque** [comment]
ci:close

**Pull request closed without merging**